components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -2514,6 +2517,11 @@ private void handleOIDCRequestObject(OAuthMessage oAuthMessage, OAuthAuthzReques
 
         validateRequestObjectParams(oauthRequest);
         String requestObjValue = null;
+        // Check whether the application mandates PAR request.
+        if (isPARMandated(parameters.getClientId(), parameters.getTenantDomain()) && !isRequestUri(oauthRequest)) {
```

Comment:
We build the par request over here https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth.par/src/main/java/org/wso2/carbon/identity/oauth/par/core/ParRequestBuilder.java#L69

If a request is a par request then this canHandler will return true https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth.par/src/main/java/org/wso2/carbon/identity/oauth/par/core/ParRequestBuilder.java#L67

Lets check on the possibility of moving the par mandatory check to https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth.par/src/main/java/org/wso2/carbon/identity/oauth/par/core/ParRequestBuilder.java#L69

One way is to check if par is mandatory check whether the Par request builder has been selected by the can handler